src/layout.jl

```diff
@@ -102,11 +102,11 @@ julia> locs_x, locs_y = spring_layout(g)
 ```
 """
 function spring_layout(g::AbstractGraph,
-                       locs_x=2*rand(nv(g)).-1.0,
-                       locs_y=2*rand(nv(g)).-1.0;
+                       locs_x_in::Vector{R1}=2*rand(nv(g)).-1.0,
+                       locs_y_in::Vector{R2}=2*rand(nv(g)).-1.0;
```

Comment:
This will break code, that passed in something else than a `Vector` - maybe it is not too big of a problem though. We could also allow at least as `AbstractVector` maybe?

Reply:
Hi @simonschoelly, What else can be passed besides a vector?

Reply:
Or we could leave this for a future PR where I plan to revisit the whole source code to make it better overall (e.g., remove the confusing all caps kwargs in `gplot`).

Reply:
I think we can change it also in `gplot`, in this direction, it is not breaking.

Reply:
done